services/welcomeEmail.js

```diff
@@ -0,0 +1,33 @@
+const sgMail = require('@sendgrid/mail');
+
+const API_KEY = process.env.API_KEY;
+const TEMPLATE_ID = process.env.TEMPLATE_ID;
+
+function sendWelcomeEmail(to, from, templateId, dynamic_template_data) {
+
+    sgMail.setApiKey(API_KEY);
+
+    const msg = {
+        to: req.body.email,
+        from: {
+            name: "ONG - Somos Más",
+            email:"somosfundacionmas@gmail.com",
+            subject: "Bievenido a la fundación Somos Más",
+        },
+        templateId: TEMPLATE_ID,
+        dynamic_template_data: {
+            subject: 'WelcomeEmailTemplate',
+            name: 'alkemyProject'
+          },
+    };
+    
+    sgMail.send(msg)
+        .then(() => {
+            console.log('Email sent')
+        })
+        .catch((error) => {
+            console.error(error)
+        })
```

Comment:
no utilizar console.log como un handler de respuesta, la funcion debe retornar algo para que el cliente de la funcion pueda ejecutar alguna accion en base al resultado.

Reply:
creé una variable emailStatus con el estado si se envió o no el mail y el status de success o error, Pero no le encontré la vuelta para que se puedan utilizar "arriba" como me indicaste. En la docu de sendgrid lo usan de esa forma con console.log:
https://docs.sendgrid.com/for-developers/sending-email/quickstart-nodejs

Como podría retornar esa variable para que la puedan usar desde el controller?

Reply:
Cambié el manejo de la respuesta usando try catch y retornando el estatus luego de ejecutar la función.